app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -217,15 +218,14 @@ def consumer_activate
       # Set it before calling find_activation_keys to allow communication with candlepin
       User.current    = User.anonymous_admin
       activation_keys = find_activation_keys
-      foreman_host    = find_foreman_host(activation_keys.first.organization)
-
-      sync_task(::Actions::Katello::System::Destroy, foreman_host.content_host) if foreman_host.try(:content_host)
+      host = Katello::Host::SubscriptionFacet.find_or_create_host(params[:facts]['network.hostname'],
```

Comment:
ohh this is actually a [new method](https://github.com/Katello/katello/blob/master/app/models/katello/host/subscription_facet.rb#L56) and not a rails method